### ISSUE TITLE
支持Gradient Checkpointing，适当降低显存需求

### ISSUE THREAD

**Issue by lzm-build** [PaddlePaddle/Paddle#62164]
### 需求描述 Feature Description

Gradient Checkpointing较梯度累积等方法可以更加显著地降低内存利用率，从 $o(n)$ 到 $o(\sqrt{n})$。有支持此功能的相关计划吗，非常感谢。

参考：
 -  https://arxiv.org/abs/1604.06174
 - https://pytorch.org/docs/stable/checkpoint.html#torch.utils.checkpoint.checkpoint
 - https://huggingface.co/docs/transformers/main_classes/model#transformers.PreTrainedModel.gradient_checkpointing_enable

**Comment by 0x45f** [comment]
你好，收到反馈，后续我们会讨论一下是否支持Gradient Checkpointing功能~
